modular_skyrat/modules/ashwalkers/code/buildings/ash_tendril.dm

```diff
@@ -0,0 +1,164 @@
+#define REQUIRED_OBSERVERS 2
+#define MEGAFAUNA_MEAT_AMOUNT 20
+
+//this is for revitalizing/preserving regen cores
+/obj/structure/lavaland/ash_walker/attackby(obj/item/attacking_item, mob/living/user, params)
+	if(!istype(attacking_item, /obj/item/organ/internal/regenerative_core))
+		return ..()
+
+	if(!user.mind.has_antag_datum(/datum/antagonist/ashwalker))
+		balloon_alert(user, "must be an ashwalker!")
+		return
+
+	var/obj/item/organ/internal/regenerative_core/regen_core = attacking_item
+	regen_core.preserved()
+	playsound(src, 'sound/magic/demon_consume.ogg', 50, TRUE)
+	balloon_alert_to_viewers("[src] revitalizes [regen_core]!")
+	return
+
+//this is for logging the destruction of the tendril
+/obj/structure/lavaland/ash_walker/Destroy()
+	var/compiled_string = "The [src] has been destroyed at [loc_name(src.loc)], nearest mobs are "
+	var/found_anyone = FALSE
+
+	for(var/mob/living/carbon/carbons_nearby in range(7))
+		compiled_string += "[key_name(carbons_nearby)],"
+		found_anyone = TRUE
+
+	if(!found_anyone)
+		compiled_string += "nobody."
+
+	log_game(compiled_string)
+	return ..()
+
+//this is for transforming a person into an ashwalker
+/obj/structure/lavaland/ash_walker/attack_hand(mob/living/user, list/modifiers)
+	if(!ishuman(user))
+		return
+
+	var/mob/living/carbon/human/human_user = user
+	if(istype(human_user.dna.species, /datum/species/lizard/ashwalker))
+		return
+
+	var/allow_transform = 0
+
+	for(var/mob/living/carbon/human/count_human in range(2, src))
+		if(!istype(count_human.dna.species, /datum/species/lizard/ashwalker))
+			continue
+		allow_transform++
+
+	if(allow_transform < REQUIRED_OBSERVERS)
+		balloon_alert_to_viewers("[src] rejects the request, not enough viewers!")
+		playsound(src, 'sound/magic/demon_consume.ogg', 50, TRUE)
+		human_user.adjustBruteLoss(10)
+		return
+
+	else
+		balloon_alert_to_viewers("[src] reaches out to [human_user]...")
+		var/choice = tgui_alert(human_user, "Become an Ashwalker? You will abandon your previous life and body.", "Major Choice", list("Yes", "No"))
+
+		if(choice != "Yes")
+			balloon_alert_to_viewers("[src] feels rejected and punishes [human_user]!")
+			playsound(src, 'sound/magic/demon_consume.ogg', 50, TRUE)
+			human_user.adjustBruteLoss(50)
+			return
+
+		balloon_alert_to_viewers("[src] rejoices and transforms [human_user]!")
+		human_user.unequip_everything()
+		human_user.set_species(/datum/species/lizard/ashwalker)
+		human_user.underwear = "Nude"
+		human_user.update_body()
```

Comment:
We ideally might want to do, like... _Something_ so they actually _look_ like an ash walker, because via just this they'll still look like Joe Nanotrasen.

Reply:
setting their species and then updating the body does change their appearance.